zcfan.c

```diff
@@ -130,9 +130,9 @@ static int write_fan(const char *command, const char *value) {
     return 0;
 }
 
-static void write_watchdog_timeout(const unsigned int timeout) {
+static void write_watchdog_timeout(const time_t timeout) {
     char timeout_s[sizeof("120")]; /* max timeout value */
-    int ret = snprintf(timeout_s, sizeof(timeout_s), "%d", timeout);
+    int ret = snprintf(timeout_s, sizeof(timeout_s), "%ld", timeout);
```

Comment:
Hmm, but this assumes it's always `long int`. That probably holds in the vast majority of cases, but simply for black box I typically use `PRIuMAX` + cast to `uintmax_t` for these cases.

Reply:
done